src/Nethermind/Nethermind.Evm/Precompiles/Stateful/BeaconBlockRootPrecompile.cs

```diff
@@ -0,0 +1,72 @@
+using System;
+using Nethermind.Core;
+using Nethermind.Core.Crypto;
+using Nethermind.Core.Specs;
+using Nethermind.Int256;
+using Nethermind.State;
+
+namespace Nethermind.Evm.Precompiles.Stateful;
+public class BeaconBlockRootPrecompile : IPrecompile<BeaconBlockRootPrecompile>
+{
+    public static Address Address { get; } = Address.FromNumber(0x0B);
+    public static UInt256 HISTORICAL_ROOTS_LENGTH = 98304;
+
+    public static BeaconBlockRootPrecompile Instance => new BeaconBlockRootPrecompile();
+
+    private byte[] SloadFromStorage(IWorldState state, in UInt256 index)
+    {
+        StorageCell storageCell = new(Address, index);
+        return state.Get(storageCell);
+    }
+    public void SetupBeaconBlockRootPrecompileState(bool isEip4788Enabled, IWorldState stateProvider, Keccak parentBeaconBlockRoot, UInt256 timestamp)
+    {
+        if (!isEip4788Enabled) return;
+        stateProvider.CreateAccountIfNotExists(BeaconBlockRootPrecompile.Address, 1);
+
+        if (parentBeaconBlockRoot is null) return;
```

Comment:
even if `parentBeaconBlockRoot is null`, we should still create the account in the line above?

Reply:
well ideally it will never be null post Cancun, so that change should be removed I think

Reply:
Also I think the nullable check then should be further up the stack not in the precompile?